smartconfig/entry.py

```diff
@@ -57,6 +57,9 @@ def _register_entry(cls) -> None:
         for node_name in cls.__path.split("."):
             current_node = current_node.setdefault(node_name, {})
 
+        if not isinstance(current_node, MutableMapping):
+            raise ConfigurationError(f"Node at path {cls.__path!r} isn't a mapping.")
+
```

Comment:
This only checks the final node. It needs to ensure all nodes along the path are mapping nodes. In fact, the above loop will fail if one of the values it retrieves does not have a `setdefault` function.

Reply:
Indeed, solved in 4ff6d5a!